pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotBrokerDebug.java

```diff
@@ -136,4 +140,21 @@ public Map<ServerInstance, List<String>> getRoutingTableForQuery(
       throw new WebApplicationException("Cannot find routing for query: " + query, Response.Status.NOT_FOUND);
     }
   }
+
+  /**
+   * API to get a snapshot of ServerRoutingStatsEntry for all the servers.
+   * @return String containing server name and the associated routing stats.
+   */
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/debug/serverRoutingStats")
+  @ApiOperation(value = "Get the routing stats for all the servers")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Server routing Stats"),
+      @ApiResponse(code = 404, message = "Server routing Stats not found"),
+      @ApiResponse(code = 500, message = "Internal server error")
+  })
```

Comment:
if someone calls this API when the stats collection is not enabled, this will NPE ?

Reply:
We will not NPE. Please see code in BaseBrokerStarter.java
serverRoutingStatsManager will always be non-null. But if the flag is disabled, we do not start any executors.